lib/core/jhipster/index.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright 2013-2020 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see http://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const _ = require('lodash');
+
+/* eslint-disable global-require */
+const builtInOptions = {
+  ...require('./application_options.js'),
+  ...require('./application_options'),
+  ...require('./application_types'),
+  ...require('./binary_options'),
+  ...require('./database_types'),
+  ...require('./default_application_options'),
+  ...require('./deployment_options'),
+  ...require('./entity_table_name_creator'),
+  ...require('./field_types'),
+  ...require('./json_entity'),
+  ...require('./relationship_options'),
+  ...require('./relationship_types'),
+  ...require('./reserved_keywords'),
+  ...require('./unary_options'),
+  ...require('./validations')
+};
+/* eslint-enable global-require */
+
+class JHipsterOptions {
+  static get OPTIONS() {
+    return _.clone(builtInOptions);
+  }
+
+  constructor(blueprintOptions) {
+    if (blueprintOptions) {
+      this.options = _.merge({}, builtInOptions, blueprintOptions);
```

Comment:
We could use destructuring instead of lodash, unless this requires some precise merging

Reply:
The merge should be precise.
```
{
  clientFramework: {
    angular: angular,
    react: react
  }
}
```
 +
```
{
  clientFramework: {
    vuejs: vuejs
  }
}
```
=

```
{
  clientFramework: {
    angular: angular,
    react: react,
    vuejs: vuejs
  }
}
```